### DISCUSSION TITLE
AWS: Centralize aws plugin auth documentation

### DISCUSSION THREAD

**Discussion by sunker** [grafana/grafana#38112] [Data Sources]
Currently, all our AWS data source plugins use the same library for handling authentication. This means they will all have the same authentication options (AWS SDK Default, Secret & keys, Credentials file, Workspace IAM) and the same configuration options. 

CloudWatch was our first AWS plugin, so all plugins that have been developed after it just reference the [authentication](https://grafana.com/docs/grafana/latest/datasources/cloudwatch/#authentication) section in the CloudWatch documentation. It's nice to have this documented in one place only, but it's also a bit weird having the CloudWatch data source being the master for this. 

What I'd like to do: 
* Create a new, generic documentation that describes how AWS data source plugin auth works. I can be based upon the authentication section in the CloudWatch documentation 
* Publish this somewhere on our website and make sure all our AWS plugins reference it in its documentation
* The only thing that is not generic about AWS plugin auth is the minimal policies for the IAM role/user required to run the plugin. That needs to be specified in the plugin itself. 

Thougts @achatterjee-grafana @aocenas @ryantxu @grafana/cloud-datasources

**Comment by achatterjee-grafana** [comment]
Hi @sunker, creating a single source of truth is a great idea, definitely less maintenance and less weird.

With regards to where we should keep this information, it will be best if we create this separate auth-only topic in markdown and keep it in our Grafana docs repository. We can single-source it (I can help with that) to show up in the CloudWatch and or other AWS topics. And we can refer to it from plugins topics living in other repos.

Question: For the newer AWS plugins, do you know if they will get classified as data sources? Or will they be similar to the redshift-data source?

**Reply by sunker** [reply]
Sounds great! 

The plan is to make all the new AWS data source plugins open source.

**Reply by Eve832** [reply]
I have OAuth info in the Cloud, if this is helpful: https://grafana.com/docs/grafana-cloud/authentication-and-permissions/

**Comment by achatterjee-grafana** [comment]
@sunker, here is the PR: https://github.com/grafana/grafana/pull/39012/. TY
